runtime/gc_structs/ConstantPoolClassSlotIterator.cpp

```diff
@@ -36,14 +36,14 @@
  * @return the next Class reference from the constant pool
  * @return NULL if there are no more references
  */
-J9Class **
+J9Class *
 GC_ConstantPoolClassSlotIterator::nextSlot()
 {
-	U_32 slotType;
-	J9Object **slotPtr;
-
-	while(_cpEntryCount) {
-		if(0 == _cpDescriptionIndex) {
+	J9Class *classPtr = NULL;
+	while (_cpEntryCount) {
+		U_32 slotType;
+		J9Object **slotPtr;
```

Comment:
they could go even further down, where they are first time initialized, but I don't know if that's up to coding standards (@dmitripivkine ?)

Reply:
Yes, I agree. Looks like there is no need for these declarations to be here. Please declare variables when they are initialized (lines 52/53)